### PR TITLE
Added get post delete edit and show for MenuTestController and tests

### LINKED ISSUE #11: Add `DELETE` endpoint for a specific record in MenuItemReview table.
Add `DELETE` endpoint for a specific record in MenuItemReview table.

# Acceptance Criteria:

- [ ] In `MenuItemReviewController.java` there is code for an 
      endpoint `DELETE /api/MenuItemReview?id=123` endpoint 
      that deletes the record if it exists, and returns 200 (ok) and 
      the text `record 123 deleted`, or returns 404 (Not Found) and
      the text `record 123 not found` if it does not.
- [ ] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [ ] The endpoint works as expected on localhost.
- [ ] The endpoint works as expected when deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the new code in 
      `MenuItemReviewController.java`
- [ ] There is full mutation test coverage (Pitest) for new code in
      `MenuItemReviewController.java`

### LINKED ISSUE #12: Add `PUT` (edit) endpoint for a single record in MenuItemReview table
Add `PUT` (edit) endpoint for a single record in MenuItemReview table

Note that it is a good idea to implement `GET` for a single item
first; when using `PUT` manually, it's convenient to be able to copy/paste the JSON representation of the data and then edit it in the swagger interface.

Eventually, it's typically the case that in order to populate an edit
form in the front end, you first do a `GET` on a particular item, 
and use that JSON to populate the fields in a form.  You can send the
JSON representation of the new item to the backend.

# Acceptance Criteria:

- [ ] In `MenuItemReviewController.java` there is code for an 
      endpoint `PUT /api/MenuItemReview?id=123` endpoint 
      that accepts JSON for a new set of values for the database
      fields other than `id`, and updates the values of those fields.
- [ ] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [ ] The endpoint works as expected on localhost.
- [ ] The endpoint works as expected when deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the new code in 
      `MenuItemReviewController.java`
- [ ] There is full mutation test coverage (Pitest) for new code in
      `MenuItemReviewController.java`

### LINKED ISSUE #13: Add `GET` (show) endpoint for a single record in MenuItemReview table
Add `GET` (show) endpoint for a single record in MenuItemReview table

# Acceptance Criteria:

- [ ] In `MenuItemReviewController.java` there is code for an 
      endpoint `GET /api/MenuItemReview?id=123` endpoint 
      that returns the JSON of the database record with id 123 if it
      exists, or a 400 and the error message `id 123 not found` if it
      does not.
- [ ] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [ ] The endpoint works as expected on localhost.
- [ ] The endpoint works as expected when deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the new code in 
      `MenuItemReviewController.java`
- [ ] There is full mutation test coverage (Pitest) for new code in
      `MenuItemReviewController.java`

### LINKED ISSUE #14: Create `MenuItemReviewController`, add `GET` (index) and `POST` (create)
Create `MenuItemReviewController`, add `GET` (index) and `POST` (create)

# Acceptance Criteria:

- [ ] There is a controller file `MenuItemReviewController.java`
      in the expected directory.
- [ ] In `MenuItemReviewController.java` there is 
      code for a `GET /api/MenuItemReview/all` endpoint 
      that returns a JSON list of all `MenuItemReview`s in the database.
      (We sometimes call this an *index* action since it lists all
      items in the database.)
- [ ] In `MenuItemReviewController.java` there is 
      code for a `POST /api/MenuItemReview/post` endpoint
      that can be used to create a new entry in the table. (This
      is a *create* action.)
- [ ] The Swagger-UI endpoints for these are well documented so that
      any member of the team can understand what they are for and
      how to use them.
- [ ] The `POST` endpoint works as expected, in the sense that new
      records can be added to the database (on localhost).
- [ ] The `GET` endpoint works as expected, in the sense that the new
      records that are added show up (on localhost).
- [ ] The `GET` and `POST` endpoints work as expected when the 
      app is deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the methods in 
      `MenuItemReviewController.java`
- [ ] There is full mutation test coverage (Pitest) for the methods in
      `MenuItemReviewController.java`

### PR DESCRIPTION
**Pull request by konman2**
Endpoints seem to work on localhost with swagger.

Closes #11 
Closes #12 
Closes #13 
Closes #14

### CODE REVIEW

**Comment by konman2** [comment]
Works on heroku.

**Review by HarvardWang — APPROVED** [review]
LGTM!

File: src/main/java/edu/ucsb/cs156/example/controllers/MenuItemReviewController.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+        MenuItemReview savedMenuItemReview = menuItemReviewRepository.save(menuItemReview);
+
+        return savedMenuItemReview;
+    }
+
+    @ApiOperation(value = "Delete a Review")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @DeleteMapping("")
+    public Object deleteUCSBDate(
```
Comment (josiahnross) [inline src/main/java/edu/ucsb/cs156/example/controllers/MenuItemReviewController.java]: Forgot to change the name from UCSB date to menu review

File: src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+                // assert
+
+                verify(menuItemReviewRepository, times(1)).findAll();
+                String expectedJson = mapper.writeValueAsString(expectedDates);
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals(expectedJson, responseString);
+        }
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void an_admin_user_can_post_a_new_ucsbdate() throws Exception {
```
Comment (josiahnross) [inline src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java]: Forgot to change the name from UCSB date to menu review

File: src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+                // assert
+                verify(menuItemReviewRepository, times(1)).save(menuItemReview1);
+                String expectedJson = mapper.writeValueAsString(menuItemReview1);
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals(expectedJson, responseString);
+        }
+
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void admin_can_delete_a_date() throws Exception {
```
Comment (josiahnross) [inline src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java]: Forgot to change the name from UCSB date to menu review

File: src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+
+                // assert
+                verify(menuItemReviewRepository, times(1)).findById(15L);
+                Map<String, Object> json = responseToJson(response);
+                assertEquals("MenuItemReview with id 15 not found", json.get("message"));
+        }
+
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void admin_can_edit_an_existing_ucsbdate() throws Exception {
```
Comment (josiahnross) [inline src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java]: Forgot to change the name from UCSB date to menu review

**Review by josiahnross — COMMENTED** [review]
Several instances where you forgot to edit the ucsb date function name to menu review

File: src/main/java/edu/ucsb/cs156/example/controllers/MenuItemReviewController.java (hunk 1 above)
Comment (konman2) [reply src/main/java/edu/ucsb/cs156/example/controllers/MenuItemReviewController.java]: updated

File: src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java (hunk 2 above)
Comment (konman2) [reply src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java]: updated

File: src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java (hunk 3 above)
Comment (konman2) [reply src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java]: updated

File: src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java (hunk 4 above)
Comment (konman2) [reply src/test/java/edu/ucsb/cs156/example/controllers/MenuItemReviewControllerTests.java]: updated

**Review by josiahnross — APPROVED** [review]
LGTM!

**Pull request merged**